### ISSUE TITLE
Marker Icon size

### ISSUE THREAD

**Issue by KeystoneScience** [react-native-maps/react-native-maps#3593]
Hello all, 
I can not find out how to change the icon size of a marker in this library. I know that you can embed an image in the marker component, however that really slows down the map as a whole, rather I am talking about sizing the icon attribute of the Marker component. 

I have tried adding styles to it with changing width and height, but this seems to have no effect. 

Any help?

Thanks!

**Comment by MichaelHettmer** [comment]
As far as I know, this is not supported at the moment. I use react-native-svg as a child component and `tracksViewChanges` is set to `false` on the markers. This makes the initial render more expensive than the icon prop, but it is still fast enough.

**Comment by KeystoneScience** [comment]
> As far as I know, this is not supported at the moment. I use react-native-svg as a child component and `tracksViewChanges` is set to `false` on the markers. This makes the initial render more expensive than the icon prop, but it is still fast enough.

Hmm. Could you provide some sample code? I have noticed that when "trackViewChanges" is set to false, the markers are never rendered with the image I put inside the marker componenet on the screen.

Here is my code: 


`                <Marker
                  key={report.couponBaseID}
                  coordinate={{
                    latitude: report.latitude,
                    longitude: report.longitude,
                  }}
                  //icon={getMarkerIcon(report.type)}
                  tracksViewChanges={mapMarkersSet}
                  onPress={(e) => {
                    e.stopPropagation();
                    console.log("Distance: ", distanceToBase(report));
                    setBaseCouponData({
                      image: report.companyURL,
                      title: report.company,
                      dealImage: report.dealURL,
                      deal: report.deal,
                      dealInfo: report.extraDealInformation,
                      numLeft: report.numLeft,
                      stopSendingOn: report.stopSendingDate,
                      expiresOn: report.expirationDate,
                      couponBaseID: report.couponBaseID,
                      locationHash: report.locationHash,
                    });
                    containerRef.current.onClickBase(report);
                  }}
                ><Image source={require('../assets/markers/treat.png')} style={{height: 40, resizeMode: "contain" }} /></Marker>
`

**Comment by MichaelHettmer** [comment]
The react-native image component is loading the image asynchronously so the first time it appears we need tracksViewChanges to be true. To improve performance we can still set tracksViewChanges to false, but only after the image has been loaded. I use onLoadEnd here, so even if loading the image fails, tracksViewChanges will be set to false.

```tsx
const MyMarker = () => {
    const [tracksViewChanges, setTracksViewChanges] = useState(true);
    return (
        <Marker tracksViewChanges={tracksViewChanges}>
            <Image src={ /* ... */ } onLoadEnd={() => setTracksViewChanges(false)} />
        </Marker>
    );
}
```

**Comment by Nicat26** [comment]
> > As far as I know, this is not supported at the moment. I use react-native-svg as a child component and `tracksViewChanges` is set to `false` on the markers. This makes the initial render more expensive than the icon prop, but it is still fast enough.
> 
> Hmm. Could you provide some sample code? I have noticed that when "trackViewChanges" is set to false, the markers are never rendered with the image I put inside the marker componenet on the screen.
> 
> Here is my code:
> 
> ` <Marker
> key={report.couponBaseID}
> coordinate={{
> latitude: report.latitude,
> longitude: report.longitude,
> }}
> //icon={getMarkerIcon(report.type)}
> tracksViewChanges={false}
> onPress={(e) => {
> e.stopPropagation();
> console.log("Distance: ", distanceToBase(report));
> setBaseCouponData({
> image: report.companyURL,
> title: report.company,
> dealImage: report.dealURL,
> deal: report.deal,
> dealInfo: report.extraDealInformation,
> numLeft: report.numLeft,
> stopSendingOn: report.stopSendingDate,
> expiresOn: report.expirationDate,
> couponBaseID: report.couponBaseID,
> locationHash: report.locationHash,
> });
> containerRef.current.onClickBase(report);
> }}
> >
> 
> <Image source={require('../assets/markers/treat.png')} style={{height: 40, resizeMode: "contain" }} />
> 
> `

It doesn't work on Android

**Comment by eyalyoli** [comment]
I checked if you can controll it by providing different img resolutions. It did on Android (Google maps) but not on iOS (Apple maps)... So there no good way around this.

**Comment by MedElba** [comment]
Hi! Why is the issue closed? Anybody found a solution? I have to display many markers with custom image loaded through https (fixed width and height, with border radius to be cirlce). I can't achieve this using the icon prop since I can't change style and dimensions, and providing the Image component inside the marker causes many performance issue!!

Any solution?

**Comment by eyalyoli** [comment]
I've updated the read me on how to do so without compromising performance...

**Comment by MedElba** [comment]
Hi @eyalyoli!

Are you talkin about this? 

Rendering a Marker with a custom image
You need to generate an png image with various resolution (lets call them custom_pin) - for more infromation go to Android, iOS
put all images in Android drawables and iOS assets dir

Because this solution doesn't work for me since i'm displaying images from internet at runtime.. 
It could help if maybe there is a way to change icon style (icon given with icon prop)

**Comment by eyalyoli** [comment]
Then your best bet is to use a custom view as a `Marker` child

**Comment by MedElba** [comment]
Yes this is what i'm doing now but with more than 20/302 markers become very very laggy!

**Comment by MedElba** [comment]
So maybe another solution would be in the native code to change the style of the icon, is there a way to achieve that?

**Comment by eyalyoli** [comment]
Yes, it is the best you can get right now. And I'm pretty sure there is another issue that is open on the performance problems. 

Loading ios/android assets from remote is a react-native bridge problem - not related to the lib.

**Comment by 0xcD3v** [comment]
Any updates guys? Am having the same issue 20+ markers the app becames slow with tracksViewChanges true, if i set the tracksViewChanges set to false the images doesn't load. The only components which render are Views and Texts components.

**Comment by apppel** [comment]
Also interested in re-opening this, using `<Image> `for a Marker is indeed much slower than, `icon=` or `image=` property.

**Comment by kkx64** [comment]
Why is this closed when this is still na issue
